scripts/entry/app.py

```diff
@@ -0,0 +1,97 @@
+from dataclasses import dataclass
+from typing import Dict
+import cfn_flip
+import json
+import subprocess
+import pathlib
+
+class Utils:
+    @staticmethod
+    def get_dict_from_template(filepath):
+        with open(filepath) as f:
+            template_raw = f.read()
+        template_flat_json = cfn_flip.to_json(template_raw)
+        template_dict = json.loads(template_flat_json)
+        return template_dict
+
+@dataclass
+class EntryTemplateFactory:
+    boilerplate: Dict
+    main:        Dict
+    entry:       Dict = None
+
+
+    @staticmethod
+    def new(**kwargs):
+
+        main_filepath = kwargs['main_filepath']
+
+        if 'boilerplate_filepath' in kwargs:
+            boilerplate_filepath = kwargs['boilerplate_filepath']
+        else:
+            boilerplate_filepath = 'input/boilerplate.yaml'
+
+        main_dict = Utils.get_dict_from_template(main_filepath)
+        boilerplate_dict = Utils.get_dict_from_template(boilerplate_filepath)
+
+
+        params = {
+            'boilerplate': boilerplate_dict,
+            'main': main_dict,
+        }
+
+        factory = EntryTemplateFactory(**params)
+        factory._update_entry(main_filepath)
```

Comment:
What if a user creates initializes this dataclass directly as is typically done with `dataclass` (instead of using your `new()` method? you could consider calling `_update_entry` as a `__post_init__()` step if it still needs to run.

e.g. user might:
```python
EntryTemplateFactory({"a": 1}, {"b": 2})
```
instead of
```python
EntryTemplateFactory.new(main_filepath="foo/bar", boilerplate_filepath="foo/baz")
```

`post_init` looks like:

```python
def __post_init__(self):
    self._update_entry(...)
```
although I guess it can't actually update_entry without the `main_filepath` param which is not a required field in initializer...

Is `dataclass` a good fit for this construct in the first place? Is it ever used as a dataclass or does it always need to be created via the "factory" method (`new()`)